File: src/templates_py/templates_py/transmitter.py
#!/usr/bin/env python3
import rclpy
from rclpy.node import Node
from std_msgs.msg import String

class Transmitter(Node):

    def __init__(self, node_name):
        super().__init__(node_name=node_name)

        self.declare_parameter(name="period", value=1.0)

        period = self.get_parameter(name="period").value

        self.pub = self.create_publisher(msg_type=String, topic='/example_topic', qos_profile=10)

        self.timer = self.create_timer(timer_period_sec=period, callback=self.publish_callback)

    def publish_callback(self):
        msg = String()
        msg.data = "music"
        self.pub.publish(msg=msg)

        self.get_logger().info(message=f'Transmitting: {msg.data}')

def main(args=None):
    rclpy.init(args=args)
    transmitter = Transmitter(node_name='transmitter')
    rclpy.spin(node=transmitter)
    rclpy.shutdown()

if __name__ == '__main__':
    main()

File: src/templates_cpp/src/transmitter.cpp
#include <rclcpp/rclcpp.hpp>
#include <std_msgs/msg/string.hpp>

class Transmitter : public rclcpp::Node
{
    public:

    Transmitter(std::string node_name) : Node(node_name)
    {
        this->declare_parameter("period", 1.0);
        period = this->get_parameter("period").as_double();
        
        pub = this->create_publisher<std_msgs::msg::String>("/example_topic", 10);
        timer = this->create_wall_timer
        (
            std::chrono::milliseconds(static_cast<int>(period*1000)),
            std::bind(&Transmitter::publish_callback, this)
        );
    }

    private:

    void publish_callback()
    {
        std_msgs::msg::String msg;
        msg.data = "news";
        pub->publish(msg);

        // RCLCPP_INFO(this->get_logger(), "Transmitting: %s", msg.data.c_str());
    }

    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr pub;
    rclcpp::TimerBase::SharedPtr timer;
    double period;
};

int main(int argc, char **argv)
{
    rclcpp::init(argc, argv);
    auto transmitter = std::make_shared<Transmitter>("transmitter");
    rclcpp::spin(transmitter);
    rclcpp::shutdown();

    return 0;
}